lib/telemetry.go

```diff
@@ -399,9 +337,25 @@ func circonusSink(cfg TelemetryConfig, hostname string) (metrics.MetricSink, err
 	return sink, nil
 }
 
+// PrometheusDefs wraps collections of metric definitions to pass into the PrometheusSink
+type PrometheusDefs struct {
+	Gauges    []prometheus.GaugeDefinition
+	Counters  []prometheus.CounterDefinition
+	Summaries []prometheus.SummaryDefinition
+}
+
+// EmptyPrometheusDefs returns a PrometheusDefs struct where each of the slices have zero elements, but not nil.
+func EmptyPrometheusDefs() PrometheusDefs {
+	return PrometheusDefs{
+		Gauges:    []prometheus.GaugeDefinition{},
+		Counters:  []prometheus.CounterDefinition{},
+		Summaries: []prometheus.SummaryDefinition{},
+	}
+}
```

Comment:
The zero value for structs and slices is usable without being initialized, so I don't think this function is necessary. The one place that uses it can use `lib.PrometheusDefs{}` which is equivalent, and avoids allocating memory for the slice headers.

Reply:
That is good to know - I was a bit concerned to pass a nil into go-metrics where I have less ability to make changes. Thankfully go does what you'd want here for once and doesn't panic w/ ranges over nil slices 😂. Seems a bit inconsistent compared to nil elsewhere, but I prefer this behavior so I'll gladly take it.